INSTALL.md

```diff
@@ -73,18 +78,32 @@ reports that Flambda is available, some further optimization options
 can be used; see the entry about `-flambda-opts` in the build guide
 for more details.
 
-Build and Installation Procedure
---------------------------------
+Choice of Build and Installation Procedure
+------------------------------------------
+
+There are two partially overlapping infrastructures available to build
+Coq. They are available through `Makefile.make` (legacy / hybrid
+build) and `Makefile.dune` (full Dune build).
+
+You can use the variable `COQ_USE_DUNE` to change which infrastructure
+to use by default. This is useful for Coq development, where we
+recommend to rely mainly on `Makefile.dune`.
+
+In both cases, the OCaml parts are built using
+[Dune](https://github.com/ocaml/dune). The main difference between the
+two infrastructures is how the `.vo` files are built.
+
+In the case of `Makefile.make`, `.vo` files are built with a legacy
+Makefile, similar to what `coq_makefile` would do. In the case of
+`Makefile.dune`, `.vo` files are built with Dune, thanks to its
+recently-added, and still experimental, Coq mode.
+
+See the documentation for the two infrastructures:
 
-Coq will build all the OCaml parts using Dune; for `.vo` files, Coq
-offers the choice of two build systems: an experimental one based on
-[Dune](https://github.com/ocaml/dune), and a makefile-based one
-[similar to `coq_makefile`].
+- [Legacy build](dev/doc/INSTALL.make.md)
+- [Full Dune build](dev/doc/build-system.dune.md)
 
-Please see [INSTALL.make.md](dev/doc/INSTALL.make.md) for build and
-installation instructions using `make`. If you wish to experiment with
-the Dune-based system see the [dune guide for
-developers](dev/doc/build-system.dune.md).
+See also [`dev/doc/README.md`](dev/doc/README.md).
```

Comment:
Looks very nice!

Optionally:
- add that an practical difference is also that `Makefile.dune` calls configuration itself
- that it is generally better not to mix the two systems [experimentally, at least due to the `_build_vo` directory and to the existence of a `config/coq_config.ml`]

Reply:
>add that an practical difference is also that Makefile.dune calls configuration itself

`./configure` is not mentioned in this file. It is in both build infrastructure documentations. (BTW, it is possible to manually call `./configure` with `Makefile.dune`, it's just not recommended for dev mode.)

>that it is generally better not to mix the two systems [experimentally, at least due to the _build_vo directory and to the existence of a config/coq_config.ml]

OK, done.